### PR TITLE
chore: organize .gitignore file

### PR DESCRIPTION
**Pull request by kkyueing**
Current `app/.gitignore` file generated by gatsby template is a little
bit too long and unorganized. This commit does two things:

- Remove unnecessary patterns
- Organize patterns by relevant groups

### CODE REVIEW

File: app/.gitignore [hunk 1]
```diff
@@ -1,69 +1,69 @@
...
-# Runtime data
+## Runtime data
 pids
 *.pid
 *.seed
 *.pid.lock
 
-# Directory for instrumented libs generated by jscoverage/JSCover
-lib-cov
+## MacOS
```
Comment (OHHAKO) [inline app/.gitignore:22]: It's okay for take the code of MacOS?

**Comment by OHHAKO** [comment]
I can't understand all of cods but i know that is for two things

File: app/.gitignore (hunk 1 above)
Comment (kkyueing) [reply app/.gitignore:22]: I guess so. Contents of the file is originally written by Gatsby's original template, and I only removed a few lines which I know for sure. There're bunch of files I don't recognize so I kept them there for the sake of safety. 😄

**Comment by kkyueing** [comment]
> I can't understand all of cods but i know that is for Organizing patterns by relevant groups

Me neither. All I know is that `.gitignore` file is a special file recognized by git, and the file makes git to ignore certain files in the project directory. For example, we usually don't want to commit or push some files such as `*.log` or `*.class`. We can write down those patterns on `.gitignore` to make git to ignore those files.

Current `.gitignore` file contains so many file patterns and I know only part of them.

**Comment by OHHAKO** [comment]
now i know exactly what '.gitignore' is. thanks for your kindness.

**Pull request merged**